pkg/archive/archive.go

```diff
@@ -169,16 +169,9 @@ func ReadTarEntry(rc io.Reader, entryPath string) (*tar.Header, []byte, error) {
 // contents will be placed. The includeRoot param sets the permissions and metadata on the root file.
 func WriteDirToTar(tw TarWriter, srcDir, basePath string, uid, gid int, mode int64, normalizeModTime, includeRoot bool, fileFilter func(string) bool) error {
 	if includeRoot {
-		rootHeader := &tar.Header{
-			Typeflag: tar.TypeDir,
-			Name:     basePath,
-			Mode:     mode,
-		}
-		if rootHeader.Mode == -1 {
-			rootHeader.Mode = int64(fs.ModePerm &^ Umask)
-		}
-		finalizeHeader(rootHeader, uid, gid, mode, normalizeModTime)
-		if err := tw.WriteHeader(rootHeader); err != nil {
+		mode = modePermIfNegativeMode(mode)
```

Comment:
```suggestion
		mode := modePermIfNegativeMode(mode)
```
(otherwise `mode` will be 0777 for child entries also... I think)

Reply:
as in the other comment, I assumed we wanted to do differently, fixed it, thanks! 😄